dsl_parser/framework/parser.py

```diff
@@ -439,6 +439,22 @@ def _calculate_element_graph(self):
                         self.element_graph.add_edge(element, dep)
                     continue
 
+                if predicates == [sibling_predicate]:
+                    # These appear to be generally in order, which means
+                    # that we can cut parsing time by a massive amount for
+                    # plugins which have a lot of DSL elements.
+                    # If we don't do this, our time complexity is n**2 as
+                    # we compare all 'default' elements to all 'type'
+                    # elements (for example), when all we care about is if
+                    # they are siblings.
+                    for dependency in dependencies:
+                        for element in dependency.parent().children():
+                            if element in _elements:
+                                self.element_graph.add_edge(element,
+                                                            dependency)
+                                assert predicates[0](element, dependency)
```

Comment:
this seems useless, surely we don't need to assert that networkx works correctly

Reply:
Oh, sorry- that was when I was testing it.

Reply:
assert removed.